### ISSUE TITLE
SigV4 / SigV4A Separation

### ISSUE THREAD

**Issue by sichanyoo** [awslabs/aws-sdk-swift#1595]
### Background

SigV4 is part of smithy spec, and belongs `smithy-swift`.
But it's in `aws-sdk-swift` currently, tangled with other things such as `SigV4Signer`.

Similarly, `awsCredentialIdentityResolver` field of client config is to be in `smithy-swift`, but it's part of `aws-sdk-swift`, along with codegen for it. 

During SRA I&A, this separation was put on hold as a backlog task to be addressed as part of the SRA modularization task down the line.

### To-do

Move all generic I&A things to `smithy-swift`, refactor runtime & codegen code where needed.

**Comment by sichanyoo** [comment]
Blocked on HTTP modularization due to circular dependency.
